### PR TITLE
Reader: Pretty URLs for some feeds

### PR DESCRIPTION
**Pull request by mjangda**
The primary goal of this PR is to redirect full site/feed URLs for Discover to the pretty version (i.e. `/discover`) but this refactors the URL generation for streams and makes it easier to add other pretty URLs in the future (`/longreads`?).

### Testing

- Verify that `/read/blog/feed/12733228` and `/read/blog/id/53424024` both redirect to `/discover`.
- Verify that all streams load correctly.
- Verify that clicking a post and going back work as expected.
- Verify that clicking on a Site Header in a stream loads that site/feed.

### CODE REVIEW

File: client/reader/controller.js [hunk 1]
```diff
@@ -19,7 +19,8 @@ var i18n = require( 'lib/mixins/i18n' ),
 	i18n = require( 'lib/mixins/i18n' ),
 	TitleStore = require( 'lib/screen-title/store' ),
 	titleActions = require( 'lib/screen-title/actions' ),
-	FeedSubscriptionActions = require( 'lib/reader-feed-subscriptions/actions' );
+	FeedSubscriptionActions = require( 'lib/reader-feed-subscriptions/actions' ),
+	urlHelper = require( 'reader/url-helper' );
```
Comment (mtias) [inline client/reader/controller.js]: Would it make sense to include this resource in `lib/route/path`? There are some similar helpers there. Or even a new `lib/route/reader.js`.

Comment (mjangda) [inline client/reader/controller.js]: I like the idea moving it to `lib/route/reader.js`. @bluefuton or @blowery any thoughts?

Comment (bluefuton) [inline client/reader/controller.js]: :+1: sounds like a good place for it to live :)

Comment (mjangda) [inline client/reader/controller.js]: Moved in 22356af and added tests in d078858

**Comment by blowery** [comment]
Maybe `reader/route` or push it into `reader/index` with the rest of the routing info, instead of `lib/route/reader`. Why would reader routing info need to break out?

**Comment by mjangda** [comment]
Cool, moved to `reader/route`.

**Comment by mtias** [comment]
@blowery I thought the file was about formatting links to use in views, in which case it makes sense as a `/lib` resource to me, since that's where we are pushing all utilities to be in. (With group sections in `client` just about views, route declarations, and controllers.) I also see there are files like `utils.js`, `x-post-helper.js`, etc in the root of reader.

**Comment by bluefuton** [comment]
Looks good to me @mjangda :+1:

@mtias @blowery do we want the view helpers to move to `lib/reader-routes` perhaps?

**Comment by mtias** [comment]
@bluefuton I don't want to hold this PR because of figuring out where to place things. We can continue the discussion in a separate issue. I do think it would make our lives easier to place all these utils in lib as a general rule. It means if we need to construct a link to the reader from somewhere else in the application we know where to look for helpers.

**Pull request merged**